regen-network/regen-web
# Feat 1350 bridge tx

## Description

Closes: regen-network/regen-registry#1350

- Wires up MsgBridge tx



<img width="991" alt="Screen Shot 2022-11-04 at 1 56 13 PM" src="https://user-images.githubusercontent.com/9438198/200063810-e13ed5ed-7877-4de6-835c-f1993cfafc1d.png">


---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...


#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

NOTE: `regen1df675r9vnf7pdedn4sf26svdsem3ugavgxmy46` has some ecocredits with batchDenom starting w/ C46.

2nd NOTE: This PR hard-codes the filter for "Bridgable" credits as `C03` (true for mainnet), but for redwood it is [`C46`](https://redwood.regen.aneka.io/txs/250F791598E6434E357089709826009019E9892292D925BB55BF05A98D62C48A) .

1. Make sure your account (`regen1df675r9vnf7pdedn4sf26svdsem3ugavgxmy46`) has some "Bridgable" ecocredits (aka "originated from polygon")
2. https://deploy-preview-1543--regen-registry.netlify.app/ecocredits/dashboard 
3. click Bridge tab
4. click Bridge button on a bridgable ecocredit
5. go through the Bridge flow (**set amount to 0.001** or any small amount until we have more to play with) and submit the TX
6. Expect success modal
7. Expect **0.001** credits to be subtracted from your "Bridgable" ecocredits
8. Expect a "BRIDGE" tx in /ecocredits/activity

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...
